scripts/eval_calc_errors.py

```diff
@@ -320,21 +350,39 @@
                             if not sphere_projections_overlap:
                                 e = [1.0] * len(p["vsd_taus"])
                             else:
-                                e = pose_error.vsd(
-                                    R_e,
-                                    t_e,
-                                    R_g,
-                                    t_g,
-                                    depth_im,
-                                    K,
-                                    p["vsd_deltas"][dataset],
-                                    p["vsd_taus"],
-                                    p["vsd_normalized_by_diameter"],
-                                    models_info[obj_id]["diameter"],
-                                    ren,
-                                    obj_id,
-                                    "step",
-                                )
+                                if p["num_workers"] == 1:
+                                    e = pose_error.vsd(
+                                        R_e,
+                                        t_e,
+                                        R_g,
+                                        t_g,
+                                        depth_im,
+                                        K,
+                                        p["vsd_deltas"][dataset],
+                                        p["vsd_taus"],
+                                        p["vsd_normalized_by_diameter"],
+                                        models_info[obj_id]["diameter"],
+                                        ren,
+                                        obj_id,
+                                        "step",
+                                    )
+                                else:  # delayed the calculation of the error for renderer_batch
+                                    e = renderer_batch.POSE_ERROR_VSD_ARGS(
+                                        R_e=R_e,
+                                        t_e=t_e,
+                                        R_g=R_g,
+                                        t_g=t_g,
+                                        depth_im=depth_im,
+                                        K=K,
+                                        vsd_deltas=p["vsd_deltas"][dataset],
+                                        vsd_taus=p["vsd_taus"],
+                                        vsd_normalized_by_diameter=p[
+                                            "vsd_normalized_by_diameter"
+                                        ],
+                                        diameter=models_info[obj_id]["diameter"],
+                                        obj_id=obj_id,
+                                        step="step",
+                                    )
 
```

Comment:
So here it would be nice to just call `pose_error.vsd(..)` with either `ren` or `renderer_batch` and then call the vsd error computation inside.